app/controllers/staff/reports_state_controller.rb

```diff
@@ -72,4 +75,8 @@ def update
   private def report
     @report ||= Report.find(params[:report_id])
   end
+
+  private def find_or_create_new_report(organisation_id:, fund_id:)
+    Report.where.not(state: :approved).find_or_create_by!(organisation_id: organisation_id, fund_id: fund_id)
```

Comment:
I didn't realise you could chain `where.not(...)` with `find_or_create_by!` like this, but I've checked the SQL and is is doing the right thing:

```sql
SELECT "reports".* FROM "reports"
WHERE "reports"."state" != 'approved'
AND "reports"."organisation_id" = '...'
AND "reports"."fund_id" = '...'
LIMIT 1;
```

If this returns nothing, it creates a new `Report` with the given `fund_id` and `organisation_id`, and relies on the database to set the default value of `state = 'inactive'`. Nice :)

Reply:
Yeah - it was new to me as well but I ❤️ it!

Reply:
I think it's a new Rails 6 one…